osu.Game/Updater/SimpleUpdateManager.cs

```diff
@@ -20,44 +15,48 @@ namespace osu.Game.Updater
     /// An update manager that shows notifications if a newer release is detected.
     /// Installation is left up to the user.
     /// </summary>
-    public partial class SimpleUpdateManager : UpdateManager
+    /// <remarks>
+    /// Implementers are expected to set <see cref="Version"/> to something that matches their release versioning/tagging system.
+    /// </remarks>
+    public abstract partial class SimpleUpdateManager : UpdateManager
     {
-        private string version;
+        protected string Version { get; set; } = "0.0.0";
 
-        [Resolved]
-        private GameHost host { get; set; }
+        protected abstract string GitHubUrl { get; }
 
-        [BackgroundDependencyLoader]
-        private void load(OsuGameBase game)
-        {
-            version = game.Version;
-        }
+        // For lack of a better term
+        protected abstract string UpdatedComponentName { get; }
```

Comment:
The lack of a better term implies overengineering. It means that it's no longer really clear what's being abstracted over.